### PR TITLE
Automate code-signing of macOS bundle

### LINKED ISSUE #5421: Sign Jenkins build artifacts to avoid security warnings
**Issue**
* macos security warnings are shown on start of application. 
* Keychain asks to re-enter password on every binary update.

**Type:** Bug.

**Solution**: Above might be solved with signing of .dmg and .app according to https://developer.apple.com/library/archive/technotes/tn2206/_index.html#//apple_ref/doc/uid/DTS40007919-CH1-TNTAG17

Makes sense to sign AppImage for Linux as well. Image creation tool supports signing via cli options.

### PR DESCRIPTION
**Pull request by gnl**
Fixes #5421 

## Review notes

The `encrypt-keychain.sh` and `gen-random-passphrase.sh` scripts are only used manually on a trusted host for the original encryption of the keychain file.

## Testing notes

* Application should open and launch without security warnings except for `App was downloaded from the internet`
* If the application was previously installed, there should be a Keychain access prompt on the first launch, but not on any subsequent launches or after future upgrades
* If the application was not previously installed, there should be no Keychain access prompts whatsoever.

### CODE REVIEW

File: deployment/macos/encrypt-keychain.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
```
Comment (vkjr) [inline deployment/macos/encrypt-keychain.sh]: When the script is called? I didn't find its mentioning among commited files.

**Review by vkjr — APPROVED** [review]

File: deployment/macos/encrypt-keychain.sh (hunk 1 above)
Comment (gnl) [reply deployment/macos/encrypt-keychain.sh]: This isn't called anywhere, it's only meant to be used manually client-side on a trusted host where the original GPG-encryption of the keychain file is performed.

Comment (vkjr) [reply deployment/macos/encrypt-keychain.sh]: Thanks for clarification :)

**Review by maxhora — APPROVED** [review]

**Review by corpetty — APPROVED** [review]

**Comment by jakubgs** [comment]
Looks like it works, nice work!

**Comment by gnl** [comment]
Squashed and rebased, ready to merge after QA confirmation.

**Comment by chadyj** [comment]
Worked for me! Yay!

Let's merge :)

**Pull request merged**